core/js/trainer.content.js

```diff
@@ -243,6 +243,93 @@ function printPokemon(pokemon,pokeimg_suffix,iv_numbers,locale){
+	else if (pokemon.cp_multiplier <= '0.706884' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '28.5';}
+	else if (pokemon.cp_multiplier <= '0.719399' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '29';}
+	else if (pokemon.cp_multiplier <= '0.719399' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '29.5';}
+	else if (pokemon.cp_multiplier <= '0.731700' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '30';}
+	else if (pokemon.cp_multiplier <= '0.731700' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '30.5';}
+	else if (pokemon.cp_multiplier <= '0.737769' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '31';}
+	else if (pokemon.cp_multiplier <= '0.737769' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '31.5';}
+	else if (pokemon.cp_multiplier <= '0.743789' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '32';}
+	else if (pokemon.cp_multiplier <= '0.743789' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '32.5';}
+	else if (pokemon.cp_multiplier <= '0.749761' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '33';}
+	else if (pokemon.cp_multiplier <= '0.749761' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '33.5';}
+	else if (pokemon.cp_multiplier <= '0.755686' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '34';}
+	else if (pokemon.cp_multiplier <= '0.755686' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '34.5';}
+	else if (pokemon.cp_multiplier <= '0.761564' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '35';}
+	else if (pokemon.cp_multiplier <= '0.761564' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '35.5';}
+	else if (pokemon.cp_multiplier <= '0.767397' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '36';}
+	else if (pokemon.cp_multiplier <= '0.767397' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '36.5';}
+	else if (pokemon.cp_multiplier <= '0.773187' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '37';}
+	else if (pokemon.cp_multiplier <= '0.773187' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '37.5';}
+	else if (pokemon.cp_multiplier <= '0.778933' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '38';}
+	else if (pokemon.cp_multiplier <= '0.778933' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '38.5';}
+	else if (pokemon.cp_multiplier <= '0.784637' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '39';}
+	else if (pokemon.cp_multiplier <= '0.784637' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '39.5';}
+	else if (pokemon.cp_multiplier <= '0.790300' && pokemon.additional_cp_multiplier <= '0') {pokemon_level = '40';}
+	else if (pokemon.cp_multiplier <= '0.790300' && pokemon.additional_cp_multiplier > '0') {pokemon_level = '40.5';}
+
```

Comment:
Not sure but if possible a formula will be better.
Anyway there is no need for 2 checks for each level just and a second if with additional_cp_multiplier for the +0.5